modules/katello_host_collection.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019, Maxim Burgerhout <maxim@wzzrd.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_host_collection
+short_description: Create and Manage Katello host collections
+description:
+    - Create and Manage Katello host collections
+author:
+    - "Maxim Burgerhout (@wzzrd)"
+requirements:
+    - "nailgun >= 0.32.0"
+    - "python >= 2.6"
+    - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    default: true
+    type: bool
+  description:
+    description:
+      - Description of the host collection
+    required: false
+  organization:
+    description:
+      - Organization that the host collection is in
+    required: true
+  name:
+    description:
+      - Name of the host collection
+    required: true
+  state:
+    description:
+      - State of the host collection
+    default: present
+    choices:
+      - present
+      - absent
+      - present_with_defaults
+'''
+
+EXAMPLES = '''
+- name: "Create Foo host collection"
+  katello_host_collection:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Foo"
+    description: "Foo host collection for Foo servers"
+    organization: "My Cool new Organization"
+    state: present
+'''
+
+RETURN = ''' # '''
+
+try:
+    from nailgun.entities import (
+        HostCollection,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_host_collection,
+        find_organization,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+    has_import_error = False
+except ImportError as e:
+    has_import_error = True
+    import_error_msg = str(e)
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'organization': 'organization',
+    'description': 'description',
+}
+
+
+def main():
+    module = AnsibleModule(
```

Comment:
in module_utils/foreman_helper, there is a `KatelloEntityAnsibleModule` to inherit from. Its use removes a lot of boilerplate here.
Also, if you feel adventurous, we started very recently to migrate the modules to a new api layer. In that case you would inherit from `KatelloEntityApipieAnsibleModule`. That is not yet required.

Reply:
Yeah I saw a few commits recently doing that. Let's move to that!

I'll change to inherit from `KatelloEntityApipieAnsibleModule` then. Anything I need to do when doing this?

Reply:
katello_content_credential is ported already. Take a look at that. It should in general become easier.

Reply:
Ok, I've inherited from Apypie now instead. Let me know if I managed to get the boilerplate right.